service/src/main/java/uk/gov/hmcts/reform/fpl/service/UserService.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.idam.client.IdamApi;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+/**
+ * Gets Users from Idam.
+ */
+@Service
+public class UserService {
+
+    private final IdamApi idamApi;
+
+    @Autowired
+    public UserService(IdamApi idamApi) {
+        this.idamApi = idamApi;
+    }
+
+    /**
+     * Makes a request to Idam and returns the user details for the user who is making the request.
+     *
+     * @param authorization the authorization token generated by the callback.
+     * @return caseLocalAuthority gained from the getEmailDomain function.
+     */
+    public String getUserDetails(String authorization) {
+        UserDetails userDetails = idamApi.retrieveUserDetails(authorization);
+        String email = userDetails.getEmail();
+        return getEmailDomain(email);
+    }
+
+    /**
+     * Removes all characters from an email address apart from the domain.
+     *
+     * @param email the email address retrieved from Idam.
+     * @return caseLocalAuthority the value to be stored in case data.
+     */
+    private String getEmailDomain(String email) {
+        int removeUser = email.indexOf("@");
+        String tempState = email.substring(removeUser + 1);
+
+        int removeEnd = tempState.indexOf(".");
+        String caseLocalAuthority = tempState.substring(0, removeEnd);
+
+        return caseLocalAuthority;
```

Comment:
We should probably lower case domain before we start processing it any further just in case it contains anything other than lower case letters.

Reply:
A good test case. I've added to lower case as follows:

```
 String tempState = email.toLowerCase().substring(start + 1);
```

Is this ok?